pkg/skeletor/build.go

```diff
@@ -15,6 +44,59 @@ package skeletor
 // Build will generate the necessary Dockerfile lines
 // for an invocation image using this mixin
 func (m *Mixin) Build() error {
+
 	//fmt.Fprintf(m.Out, dockerfileLines)
+
+	// ======================================================================================
+	// REMOVE THIS CODE BELOW IF YOUR MIXIN DON\'T REQUIRE A CLIENT VERSION
+	// =====================================  START =========================================
+	// Create new Builder.
+	var input BuildInput
+
+	err := builder.LoadAction(m.Context, "", func(contents []byte) (interface{}, error) {
+		err := yaml.Unmarshal(contents, &input)
+		return &input, err
+	})
+	if err != nil {
+		return err
+	}
+
+	suppliedClientVersion := input.Config.ClientVersion
+
+	if suppliedClientVersion != "" {
```

Comment:
I might suggest we remove the semver constraint.  I'm thinking for the most cases of mixins, semver restrictions won't be needed and may be prove too restrictive. WDYT?

Reply:
Yes, you are right! semver restrictions are too much and you are right also about the comment/uncomment code, looks pretty noisy. There is another way to generate the desired code, but we have to use `https://github.com/cookiecutter/cookiecutter`  and this mean this project  would be converted into `cookiecutter` style, and then we could much more flexibility of generating an initial mixin code

Reply:
The cookiecutter project looks interesting and perhaps worth revisiting in the future.

I think what we have now with your updates is great: support/example for processing mixin config, with clientVersion as a great standard entry.  I think if we add additional context/comments around the `//fmt.Fprintf(m.Out, dockerfileLines)` line, maybe showing an example that uses `m.ClientVersion`, we should be good to go.